src/sage/combinat/symmetric_group_algebra.py

```diff
@@ -1941,6 +1944,57 @@ def _dft_seminormal(self, mult='l2r'):
         snb = self.seminormal_basis(mult=mult)
         return matrix([vector(b) for b in snb]).inverse().transpose()
 
+    def _dft_modular(self):
+        r"""
+        Return the discrete Fourier transform when the characteristic divides the order of the group.
+
+        See [Mur1983]_ for the construction of central primitive orthogonal idempotents.
+        For each idempotent `e_i` we have a homomorphic projection `v \mapsto v e_i`.
+        Choose a basis for each submodule spanned by `\{\sigma e_i | \sigma \in S_n\}`.
+        The change-of-basis from the standard basis `\{\sigma\}_\sigma` is returned.
+
+        EXAMPLES::
+
+            sage: GF2S3 = SymmetricGroupAlgebra(GF(2),3)
+            sage: GF2S3.dft(form="modular")
+            [1 0 0 0 1 0]
+            [0 1 0 0 0 1]
+            [0 0 1 0 0 1]
+            [0 0 0 1 1 0]
+            [1 0 0 1 1 0]
+            [0 1 1 0 0 1]
+
+        Check that :issue:`37751` is fixed::
+            
+            sage: GF3S3 = SymmetricGroupAlgebra(GF(3),3)
+            sage: GF3S3.dft(form="modular")
+            [1 0 0 0 0 0]
+            [0 1 0 0 0 0]
+            [0 0 1 0 0 0]
+            [0 0 0 1 0 0]
+            [0 0 0 0 1 0]
+            [0 0 0 0 0 1]
+        """
+        #create a spanning set for the block corresponding to an idempotent
+        spanning_set = lambda idem: [self(sigma)*idem for sigma in self.group()]
+        #compute central primitive orthogonal idempotents
+        idempotents = self.central_orthogonal_idempotents()
+        #project v onto each block U_i = F_p[S_n]*e_i via \pi_i: v |--> v*e_i
+        blocks = [self.submodule(spanning_set(idem)) for idem in idempotents]
+        #helper function to flatten a list
+        from sage.misc.flatten import flatten
+        #compute the list of basis vectors lifed to the SGA from each block
+        block_decomposition_basis = flatten([[u.lift() for u in block.basis()] for block in blocks])
+        #the elements of the symmetric group are ordered, giving the map from the standard basis
+        sym_group_list = list(self.group())
+        change_of_basis_matrix = []
+        for b in block_decomposition_basis:
+            coord_vector = [0]*len(sym_group_list)
+            for pair in list(b):
+                coord_vector[sym_group_list.index(pair[0])] = pair[1]
+            change_of_basis_matrix.append(coord_vector)
+        return matrix(self.base_ring(),change_of_basis_matrix).transpose()
```

Comment:
```suggestion
        # construct the matrix to the standard basis in the order given by the group
        G = self.group()
        mat = [[b[g] for b in block_decomposition_basis] for g in G]
        return matrix(self.base_ring(), mat)
```
If I got the transposed matrix, then use
```
        mat = [[b[g] for g in G] for b in block_decomposition_basis]
```